LFS-360: Clarify the source of filter variables

OrdiNeu opened the pull request:
Change the filters so that they have the questionnaire before them, so it lets you know which questionnaire you're pulling data from,

The UI might need a rework.

![image](https://user-images.githubusercontent.com/4656440/123850933-98a69b80-d8e8-11eb-84c1-7aabbcb8f287.png)

OrdiNeu committed:
LFS-360: Clarify the source of filter variables

OrdiNeu committed:
LFS-360: Clarify the source of filter variables
Go back to using components instead of react native.

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/tableStyle.jsx:
```diff
@@ -55,6 +55,10 @@ const liveTableStyle = theme => ({
+        padding: theme.spacing(0, 2),
+    },
+    categoryHeader: {
+        backgroundColor: "#FFFFFF"
```
```suggestion
        backgroundColor: theme.palette.background.paper
```

OrdiNeu committed:
LFS-360: Clarify the source of filter variables

Apply code review comments by @marta-

Co-authored-by: Marta Girdea <marta.girdea@gmail.com>

sashaandjic commented:
![image](https://user-images.githubusercontent.com/23243244/123981709-77998580-d990-11eb-9f8c-28a4ed0739fb.png)

OrdiNeu committed:
[Misc] Fix merge conflict-caused bug

OrdiNeu committed:
LFS-360: Clarify the source of filter variables
Fix a bug where boolean questions weren't being filtered properly

OrdiNeu commented:
@sashaandjic 
> Clicked on `+` to add a filter and got this: 

Should be fixed in `b778576` :heavy_check_mark:

sashaandjic commented:
Can we make `Patient information`, `Radiotherapy`, ... entries not clickable? Larger font? Would any kind indentation be possible here?
![image](https://user-images.githubusercontent.com/23243244/124717397-14997880-ded3-11eb-80e2-4b6ba824422a.png)

Ideally, it should be similar to the vocabulary browser (without `i` buttons). This is surely a bit of a stretch. 
![image](https://user-images.githubusercontent.com/23243244/124718770-6ee70900-ded4-11eb-82d3-e90b33d48f4f.png)

OrdiNeu commented:
> Can we make `Patient information`, `Radiotherapy`, ... entries not clickable? Larger font? Would any kind indentation be possible here?

They shouldn't be clickable currently -- what happens when you click on them? 

> Ideally, it should be similar to the vocabulary browser (without `i` buttons). This is surely a bit of a stretch. 

I can try my hand at restyling this.

OrdiNeu committed:
LFS-360: Clarify the source of filter variables
Styling fixes

marta- committed:
LFS-360: Clarify the source of filter variables
Minor styling improvement

marta- committed:
LFS-360: Clarify the source of filter variables
Minor styling improvement

sashaandjic commented:
Looks very good!
I am unable to select "Created Date" filter. Not sure if this is within the scope of this PR.
Clicking on "Patient Information",  "Tissue Matrix", etc. still closes the list. I would expect nothing to happen when clicking on these.
![image](https://user-images.githubusercontent.com/23243244/125145956-9e269180-e0f1-11eb-93be-c7e68a123440.png)

marta- commented:
> I am unable to select "Created Date" filter. Not sure if this is within the scope of this PR.

I confirm the problem. It is within the scope, looks like a regression.

> Clicking on "Patient Information", "Tissue Matrix", etc. still closes the list. I would expect nothing to happen when clicking on these.

Both Francis and I explored several ways to prevent it from closing to no avail. I'm giving up for now, let's report it as an improvement for later.

marta- commented:
Another bug:
* Open the filters dialog
* Add new filter
* Select "Date of birth" from "Patient information"
* Click on the drop-down again
* Notice that the selected item is actually "Date of birth" from "Tissue Metric"

sashaandjic commented:
Confirmed the bug reported above. 
Also, can we make "Add new filter..." more visible.  Blue fonts as in "Patient information" would look nice.
![image](https://user-images.githubusercontent.com/23243244/125148296-3d9e5100-e0ff-11eb-8d74-ac1a9f120481.png)

sdumitriu commented:
I don't think we need to make it more visible. I would remove it actually.

marta- committed:
LFS-360: Clarify the source of filter variables
Prevent the select from closing when the questionnaire titles are clicked: replaced `ListSubheader` with a disabled `MenuItem`

marta- committed:
LFS-360: Clarify the source of filter variables
Prevent the select from closing when the questionnaire titles are clicked: replaced `ListSubheader` with a disabled `MenuItem`

marta- commented:
> Clicking on "Patient Information", "Tissue Matrix", etc. still closes the list. I would expect nothing to happen when clicking on these

Managed to fix this in f50e7b1.

marta- commented:
> Also, can we make "Add new filter..." more visible. Blue fonts as in "Patient information" would look nice.

@sashaandjic did you want it looking more like a button?

Uppercase:

![image](https://user-images.githubusercontent.com/651980/125150424-b73d3b80-e10d-11eb-96dc-4f8625bd3b3b.png)
![image](https://user-images.githubusercontent.com/651980/125150446-d20fb000-e10d-11eb-8bec-94a4cac9a867.png)

No uppercase:
![image](https://user-images.githubusercontent.com/651980/125150474-169b4b80-e10e-11eb-8713-753e14bf54ce.png)
![image](https://user-images.githubusercontent.com/651980/125150468-fff4f480-e10d-11eb-9e1b-a3a4ff6fde92.png)

sashaandjic commented:
> > Clicking on "Patient Information", "Tissue Matrix", etc. still closes the list. I would expect nothing to happen when clicking on these
> 
> Managed to fix this in [f50e7b1](https://github.com/ccmbioinfo/lfs/commit/f50e7b1d02b68d9750163d6d8c426e2405e5e8b8).

Tested, I noticed that the questionnaires are no longer bolded and blue. Was that done intentionally?

sashaandjic commented:
> @sashaandjic did you want it looking more like a button?

Ah, now I see why it is displayed in gray fonts. Consistency.
I actually like shaded "Add new filter..." in the drop down. It is as an fyi and does not confuse user. If we go with blue and bold, it may be too much and user may get confused, However in the list it just looks odd being shaded. @sdumitriu  suggested removing it from the list. WDYT?
I also liked it better with blue and bolded fonts in "Patient information".
![image](https://user-images.githubusercontent.com/23243244/125152281-98dd3d00-e119-11eb-802b-9d963911556e.png)

marta- commented:
> Tested, I noticed that the questionnaires are no longer bolded and blue. Was that done intentionally?

They are blue, I just forgot to make them bold again. Fixed in 60a476e.

marta- committed:
LFS-360: Clarify the source of filter variables
Restore bold font for questionnaire titles

marta- commented:
> I also liked it better with blue and bolded fonts in "Patient information". Can we add a little dot before field names in the list? Or maybe ":" after "Patient information"?

I think indentation is clear enough, bullets and punctuation would make everything look a bit too crowded.

marta- commented:
> @sdumitriu suggested removing it from the list. WDYT?

Is what you had in mind? I'm not a fan of how abruptly the list starts and I would leave it unchanged.

![image](https://user-images.githubusercontent.com/651980/125165158-c5b64200-e163-11eb-931f-031c9ae40380.png)

marta- commented on modules/data-entry/src/main/frontend/src/dataHomepage/Filters.jsx:
```diff
@@ -317,6 +372,21 @@ function Filters(props) {
+    return fields.map((name) => {
+      if (typeof name == "string") {
+        // Straight strings are MenuItems
+        return <MenuItem value={name} key={name} className={classes.categoryOption + (nested ? " " + classes.nestedSelectOption : "")}>{filterableTitles[name]}</MenuItem>
```
We should probably use `value={path}` instead of the `name` to avoid collisions. This is in relation with the bug where you select "Date of birth" in "Patient information", and "Date of Birth" also gets selected in "Tissue Metrix".

OrdiNeu replied:
I've renamed it in `55debe2` and did the change from name->path in `802e98a` :heavy_check_mark:

marta- commented:
@OrdiNeu Summary of current issues:
- [ ] **Regression:** "Created Date" filter cannot be selected, and a js console error is generated: `DateFilter.jsx:100 Uncaught TypeError: Cannot read property 'dataType' of undefined`.
- [ ] **Bug**: Selecting "Date of birth" from "Patient information" also makes "Date of birth" from "Tissue Metrix" appear selected. Same issue with "Life status".

OrdiNeu committed:
LFS-360: Clarify the source of filter variables

Fix an issue with Subjects and CreatedDate not being selectable
Fix an issue where the same question under multiple questionnaires would
cause you to be unable to select a filter.
Fix a console warning about defaultLabel being applied to TextField

OrdiNeu committed:
LFS-360: Clarify the source of filter variables

Switch the variable "name" with "path" to better reflect what it is

OrdiNeu commented:
@marta- 
> @OrdiNeu Summary of current issues:

Fixed in `802e98a` :heavy_check_mark:

sashaandjic reviewed (approved):
tested

sdumitriu commented on modules/data-entry/src/main/frontend/src/dataHomepage/Filters.jsx:
```diff
@@ -81,36 +82,94 @@ function Filters(props) {
+      newFilterableFields.push([title, ...parseSectionOrQuestionnaire(thisQuestionnaire)]);
+    }
+
+    newQuestionDefinitions["Subject"] = {
+      dataType: "subject"
+    };
```
Isn't this already initialized above on line 120?
